### PR TITLE
crc and hex funcs

### PR DESCRIPTION
**Pull request by bheni**

### CODE REVIEW

File: sql/expression/function/math.go [hunk 1]
```diff
@@ -113,282 +114,244 @@ func (r *Rand) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
...
+			return nil, err
+		}
 
-// EvalFloat handles the case where the number argument passed is a float
-func (fl CotFuncLogic) EvalFloat(n float64) (interface{}, error) {
-	return 1.0 / math.Tan(n), nil
-}
+		a := n.(int64)
+		if a < 0 {
+			bytes := *(*[8]byte)(unsafe.Pointer(&a))
```
Comment (zachmu) [inline sql/expression/function/math.go:328]: Might add a link to some docs describing this algorithm. It's kinda magic

File: sql/expression/function/absval_test.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	}
+
+	testCases := []struct {
+		name     string
+		typeToConv    map[sql.Type]toTypeFunc
+		val      float64
+		expected float64
+		err      error
+	}{
+		{
```
Comment (zachmu) [inline sql/expression/function/absval_test.go:55]: Need tests of behavior for nil values

File: sql/expression/function/math_test.go [hunk 3]
```diff
@@ -234,26 +119,102 @@ func TestDegrees(t *testing.T) {
...
 	}
 }
 
+func TestCRC32(t *testing.T) {
+	tests := []struct{
+		name string
+		input interface{}
+		expected uint32
+	}{
+		{"CRC32('MySQL)", "MySQL", 3259397556},
```
Comment (zachmu) [inline sql/expression/function/math_test.go:140]: Needs tests for nil values

File: sql/expression/function/math_test.go [hunk 4]
```diff
@@ -234,26 +119,102 @@ func TestDegrees(t *testing.T) {
...
+	}
+}
+
+func TestHexFunc(t *testing.T) {
+	tests := []struct{
+		name string
+		input interface{}
+		expected string
+	}{
+		{"HEX('test')", "test", "74657374"},
```
Comment (zachmu) [inline sql/expression/function/math_test.go:178]: Need nil tests

File: sql/expression/function/math.go (hunk 1 above)
Comment (bheni) [reply sql/expression/function/math.go:328]: refactored and commented.

File: sql/expression/function/absval_test.go (hunk 2 above)
Comment (bheni) [reply sql/expression/function/absval_test.go:55]: done

File: sql/expression/function/math_test.go (hunk 3 above)
Comment (bheni) [reply sql/expression/function/math_test.go:140]: done
